src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java

```diff
@@ -0,0 +1,79 @@
+package org.openmrs.isanteplus.performancedata.model;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Getter
+@Setter
+public class Encounter extends AbstractEntity {
+
+    private long encounterType;
+
+    private long patientId;
+
+    private long encounterId;
+
+    private long locationId;
+
+    private long formId;
+
+    private LocalDateTime encounterDatetime;
+
+    private long creator;
+
+    private LocalDateTime dateCreated;
+
+    private long voided;
+
+    private long voidedBy;
+
+    private LocalDateTime dateVoided;
+
+    private String voidReason;
+
+    private long changedBy;
+
+    private LocalDateTime dateChanged;
+
+    private long visitId;
+    
+    private UUID uuid;
+
+    public Encounter() {
+        preparedSql = "insert into encounter"
+                + " (encounter_id, encounter_type, parent_id, location_id, form_id, " +
+                "encounter_datetime, creator, date_created, voided, voided_by, date_voided, " +
+                "void_reason, changed_by, date_changed, visit_id, uuid)"
+                + " values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+    }
+
+    @Override
+    public PreparedStatement fillPreparedStatement(PreparedStatement ps) throws SQLException {
+        ps.setString(1, Long.toString(encounterType));
```

Comment:
This is pretty ugly, can't we just use Spring JDBC for this? 

https://www.mkyong.com/spring/spring-named-parameters-examples-in-simplejdbctemplate/ - Example 2

With a BeanPropertySqlParameterSource you can drop the whole prepare method and replace the question marks by property names like :name, :encounterType, etc.